### PR TITLE
[Compute] add error message for `vm identity assign`

### LINKED ISSUE #17682: [Compute] Pointless error message in `vm identity assign`
As #17619 

**Command Name**
`az vm identity assign`

If cloud profile is old, CLI will report a NoneType error
```
'NoneType' object is not callable
Traceback (most recent call last):
python3.8/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 711, in _run_job
    six.reraise(*sys.exc_info())
...
cli/command_modules/vm/custom.py, ln 648, in assign_vm_identity
    assign_identity_helper(cmd.cli_ctx, getter, setter, identity_role=identity_role_id, identity_scope=identity_scope)
cli/core/commands/arm.py, ln 1057, in assign_identity
    resource = setter(resource)
cli/command_modules/vm/custom.py, ln 642, in setter
    vm.identity.user_assigned_identities[identity] = VirtualMachineIdentityUserAssignedIdentitiesValue()
TypeError: 'NoneType' object is not callable
```

### PR DESCRIPTION
**Pull request by xfz11**
**Description**

resolve #17682 
**command**:
`az vm identity assign`

**result**
Before:
```
'NoneType' object is not callable
Traceback (most recent call last):
python3.8/site-packages/knack/cli.py, ln 215, in invoke
    cmd_![image](https://user-images.githubusercontent.com/81600993/114687941-d0209580-9d46-11eb-8237-6365b365ea73.png) = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 711, in _run_job
    six.reraise(*sys.exc_info())
...
cli/command_modules/vm/custom.py, ln 648, in assign_vm_identity
    assign_identity_helper(cmd.cli_ctx, getter, setter, identity_role=identity_role_id, identity_scope=identity_scope)
cli/core/commands/arm.py, ln 1057, in assign_identity
    resource = setter(resource)
cli/command_modules/vm/custom.py, ln 642, in setter
    vm.identity.user_assigned_identities[identity] = VirtualMachineIdentityUserAssignedIdentitiesValue()
TypeError: 'NoneType' object is not callable
```
After
image

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -670,6 +671,10 @@ def setter(vm, external_identities=external_identities):
         vm.identity = VirtualMachineIdentity(type=identity_types)
         if external_identities:
             vm.identity.user_assigned_identities = {}
+            if not cmd.supported_api_version(min_api='2018-06-01', resource_type=ResourceType.MGMT_COMPUTE):
+                raise CLIInternalError("usage error: user assigned identity is not available under current profile.",
+                                       "Please set the cloud's profile to latest with 'az cloud set --profile latest"
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:676]: Profiles are not static. Other profiles may support it in the future too.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -670,6 +671,10 @@ def setter(vm, external_identities=external_identities):
         vm.identity = VirtualMachineIdentity(type=identity_types)
         if external_identities:
             vm.identity.user_assigned_identities = {}
+            if not cmd.supported_api_version(min_api='2018-06-01', resource_type=ResourceType.MGMT_COMPUTE):
+                raise CLIInternalError("usage error: user assigned identity is not available under current profile.",
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:675]: A sentence should start with a capital letter.
Add a space between two sentences.

Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:675]:
```suggestion
                raise CLIInternalError("Usage error: user assigned identity is not available under current profile.",
```

Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:675]:
```suggestion
                raise CLIInternalError("usage error: User assigned identity is not available under current profile. ",
```

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:676]:
```suggestion
                                       "You can set the cloud's profile to latest with 'az cloud set --profile latest"
```

**Review by qwordy — APPROVED** [review]

**Pull request merged**